src/test/Modal.test.tsx

```diff
@@ -0,0 +1,33 @@
+import '@testing-library/jest-dom'
+import { shallow, mount } from 'enzyme'
+import { Modal } from '../components/modal'
+import { IModalProps } from '../components/modal/IModalProps'
+
+const setShowModal = jest.fn()
+const defaultProps : IModalProps = {
+    setShowModal,
+    title: '',
+    footer: ''
+}
+
+describe('Testing modal component', () => {
```

Comment:
We would recommend no longer using Enzyme and switching to React testing library for this kind of tests. Enzyme does not support React 17 officially yet (and React 18 is going to be released sooner than later).

With RTL you can do this tests without mocking the `setShowModal`, which, as I said before is an odd prop.

Reply:
Okay, I'll figure it out and implement the tests using just React testing library. I was used to use [this adapter](https://github.com/wojtekmaj/enzyme-adapter-react-17) (which is unofficial). I will no longer use it then!

Reply:
I migrated the tests using [this docs page](https://testing-library.com/docs/react-testing-library/migrate-from-enzyme)